### PR TITLE
For #128: Joined validation pattern

### PR DESCRIPTION
**Pull request by paulodamaso**
For #128:
- created joined validation pattern
- added tests

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#203](https://github.com/yegor256/aibolit/issues/203) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/e6bce1f0-b1a0-4a21-a44a-e763f8df8e8e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: test/patterns/joined_validation/test_joined_validation.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from unittest import TestCase
+from aibolit.patterns.joined_validation.joined_validation import JoinedValidation
+from pathlib import Path
+
+
+class TestJoinedValidation(TestCase):
+    dir_path = Path(os.path.realpath(__file__)).parent
+
+    def test_simple(self):
+        self.assertEqual([9], JoinedValidation(Path(self.dir_path, 'JoinedValidation.java')).value())
```
Comment (lyriccoder) [inline test/patterns/joined_validation/test_joined_validation.py:12]: Check also such cases:
   
```
 if (a == 1 && x == 1 || y == 1) { // here!
            System.out.println("Not here, also");
        }
```

```
 if (x == 1 || y == 1 && a == 1) { // here!
            System.out.println("Not here, also");
        }
```
```
if (x == 1 && y == 1) { // not here!
            if (true)
               throw new Exception("Oops");
        }
```
```
        if (x == 1 && y == 1) { // not here!
            System.out.println("Not here, also");
            throw new Exception("Oops");
        }
```
```
        if (x == 1 || y == 1) {
        }
```

[the remark this answers is not in the document]
Comment (lyriccoder) [reply test/patterns/joined_validation/test_joined_validation.py:12]: Check cases:
```
 if (a == 1 && x == 1 || y == 1) {
            throw new Exception("Oops");
        }
```
```
 if (x == 1 || y == 1 && a == 1) { 
            throw new Exception("Oops");
        }
```
```
if (x == 1 && y == 1) { 
            if (true)
               throw new Exception("Oops");
        }
```
```
        if (x == 1 || y == 1 || y == 1) { 
            throw new Exception("Oops");
        }
```
```
        if (x == 1 || fun.call() == 1) {
          throw new Exception("Oops");
        }
```
```
        if (x == 1 ||  this.a != 1) {
          throw new Exception("Oops");
        }
```

**Comment by acheshkov** [comment]
@paulodamaso Are we ready to merge?

**Comment by paulodamaso** [comment]
@lyriccoder Done, please take a look

**Comment by paulodamaso** [comment]
@acheshkov @lyriccoder asked me to fix some issues, I did it on the last commit, now he have to review it

**Comment by lyriccoder** [comment]
Such case has an exception:

```
class SimpleJoinedValidation {
    void print(int x, int y) {
        if (x == 1 || y == 1)
            throw new Exception("Oops");
    }
}
```

**Comment by paulodamaso** [comment]
@lyriccoder All test cases added, please take a look

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**